src/errors.rs

```diff
@@ -370,7 +341,168 @@ error_chain! {
     }
 }
 
-fn valid_profile_names() -> String {
+#[derive(ThisError, Debug)]
+pub enum RustupError {
+    #[error("invalid toolchain name: '{0}'")]
+    InvalidToolchainName(String),
+    #[error("Unable to proceed. Could not locate working directory.")]
+    LocatingWorkingDir {
+        #[from]
+        source: io::Error,
+    },
+    #[error("failed to set permissions for '{}'", .p.display())]
```

Comment:
```suggestion
    #[error("failed to set permissions for '{p}'")]
```

We'll display the path automatically.